### DISCUSSION TITLE
ESP32-S2 , MAX98357, Volume always at maximum

### DISCUSSION THREAD

**Discussion by musiq021** [pschatzmann/arduino-audio-tools#958] [Q&A] [answered]
I saw someone else posted almost the exact issue, unfortunately their issue was they simply weren't routing through the VolumeStream. 

I am routing through the stream, as evidenced by the audio stopping entirely when I setVolume to 0. But anything between 0 and 1 results in no change / full volume.

**Comment by pschatzmann** [comment, marked as answer]
I just created a [test case](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/tests/test-volumestream/test-volumestream.ino) and it's working perfectly. 

Maybe you just forgot to configure the VolumeStream class ?

I suggest that you also follow these recommendations: https://github.com/pschatzmann/arduino-audio-tools/wiki/It's-not-working

**Reply by musiq021** [reply]
Thank you for the incredibly fast sanity check. Im a novice and didnt realize the L/R input on the module was WS / word select. I didnt expect it would work at all if the wiring was wrong. Working well now
